src/EditProfile.js

```diff
@@ -0,0 +1,174 @@
+import React from "react";
+import "./profile.css"
+import {withRouter, Link} from 'react-router-dom'
+
+  class EditProfile extends React.Component {
+    constructor(props) {
+      super(props);
+      this.state = {
+        userId: this.props?.location?.state?.id,
+        name: "",
+        age: "null",
+        bio: "",
+        gender: "Female",
+        genderPreference: "Female",
+        education: "",
+        interests: "",
+        error: "",
+        maxDistance: 10,
+        updateStatus: "",
+      };
+
+      this.handleUpdate = this.handleUpdate.bind(this);
+      this.handleChange = this.handleChange.bind(this);
+    }
+    componentDidMount(){
+        const myRequest = new Request('http://127.0.0.1:5000/viewprofile', {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+            body: JSON.stringify({"userId": this.state.userId}),
+            });
+            fetch(myRequest)
+                .then(response => response.json())
+                .then(res => 
+                        this.setState({
+                            name: res.name,
+                            age: res.age, 
+                            bio: res.bio,
+                            gender: res.gender,
+                            genderPreference: res.genderPreference,
+                            education: res.education, 
+                            interests: res.interests,
+                            maxDistance: res.maxDistance,
+                        })
+            ).catch((error) => {
+                console.error(error)
+        })
+      }
+
+    handleUpdate = (event) => {
+      event.preventDefault();
+      if(this.completedInput()) {
+        const id = this.props?.location?.state?.id;
+        const myForm = new FormData (document.getElementById("profileForm"));
+        myForm.append("id", id);
+        const myRequest = new Request("http://127.0.0.1:5000/editprofile", {
+          method: "POST",
+          body: myForm,
+        });
+
+        fetch(myRequest)
+        .then((res) =>
+          res.json())
+        .then((res) => {
+          if(res.response === "Success"){
```

Comment:
Would it be necessary to write a function to check if there are changes in the profile before submitting the form? Or even if there is no change, we will still update the profile with what it's currently showing?